File: 125-valid-palindrome/valid-palindrome.py
import string

class Solution:
    def isPalindrome(self, s: str) -> bool:
        word =s 
        i = 0
        j = len(word) - 1

        while i < j:

            if ord(word[i].lower()) not in range(97, 123) and ord(word[i].lower()) not in range(48, 58):
                i += 1
            elif ord(word[j].lower()) not in range(97, 123) and ord(word[j].lower()) not in range(48, 58):
                j -= 1
            elif ord(word[i].lower()) != ord(word[j].lower()):
                return False
            else:
                i += 1
                j -= 1
        return True
        

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String word) {
        int i = 0, j = word.length() - 1;
        while (i < j) {
            char a = Character.toLowerCase(word.charAt(i));
            char b = Character.toLowerCase(word.charAt(j));
            if (!isAlnum(a)) {
                i++;
            } else if (!isAlnum(b)) {
                j--;
            } else if (a != b) {
                return false;
            } else {
                i++;
                j--;
            }
        }
        return true;
    }

    private boolean isAlnum(char c) {
        return (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9');
    }
}